aptos-move/block-executor/src/executor.rs

```diff
@@ -1029,6 +1020,24 @@ where
             let res = executor.execute_transaction(&latest_view, txn, idx as TxnIndex);
             let must_skip = matches!(res, ExecutionStatus::SkipRest(_));
             match res {
+                ExecutionStatus::Abort(err) => {
+                    if let Some(commit_hook) = &self.transaction_commit_hook {
+                        commit_hook.on_execution_aborted(idx as TxnIndex);
+                    }
+                    alert!("Fatal VM error by transaction {}", idx as TxnIndex);
+                    // Record the status indicating the unrecoverable VM failure.
+                    return Err(SequentialBlockExecutionError::ErrorToReturn(BlockExecutionError::FatalVMError(err)));
+                },
+                ExecutionStatus::DelayedFieldsCodeInvariantError(msg) => {
+                    alert!("Discarding transaction because we got DelayedFieldsCodeInvariantError in the sequential fallback: {msg}");
+                    ret.push(E::Output::discard_output(StatusCode::DELAYED_MATERIALIZATION_CODE_INVARIANT_ERROR));
+                    continue;
+                },
+                ExecutionStatus::SpeculativeExecutionAbortError(msg) => {
+                    alert!("Discarding transaction because we got SpeculativeExecutionAbortError in the sequential fallback {msg}");
+                    ret.push(E::Output::discard_output(StatusCode::DELAYED_MATERIALIZATION_CODE_INVARIANT_ERROR));
```

Comment:
a different argument here, but also for considering block error instead of discarding would be that this should be so impossible during sequential execution that one could almost advocate unreachable or skipping a block vs discarding an individual transaction.